Sources/Packages.php

```diff
@@ -1026,6 +1026,10 @@ function PackageInstall()
 			}
 			elseif ($action['type'] == 'hook' && isset($action['hook'], $action['function']))
 			{
+				// Set the system to ignore hooks, but only if it wasn't changed before.
+				if (!isset($context['ignore_hook_errors']))
```

Comment:
Looks good, just curios about this, does this mean that if a mod doesn't want to ignore errors the mod author needs to explicitely set it to false?  if so, wouldn't it be better to define and set to true this key before any hook is run to make it the default behaviour?

Reply:
Well that would be possible.  But this is only being set during package install if a hook is used.
Technically it means another part of SMF could trigger this as well or another mod.

This is purposely handling just the hook handling where as mentioned the reloadSettings is called during the package install and then due to how other parts of SMF are executed afterwords, hooks may get called too early on, or without other needed hooks.

Reply:
Ok cool!